server/src/main/java/com/cloud/api/ApiServer.java

```diff
@@ -874,6 +875,34 @@ private void buildAuditTrail(final StringBuilder auditTrailSb, final String comm
         }
     }
 
+    protected boolean verifyApiKeyAccessAllowed(User user, Account account) {
+        Boolean apiKeyAccessEnabled = user.getApiKeyAccess();
+        if (apiKeyAccessEnabled != null) {
+            if (apiKeyAccessEnabled == true) {
+                return true;
+            } else {
+                logger.info("Api-Key access is disabled for the User");
```

Comment:
Please log the user details, UUID and name in all the places

Reply:
Done